django_api/restaurants/views.py

```diff
@@ -1,3 +1,51 @@
-from django.shortcuts import render
+# from django.http import HttpResponse
+# import random
+# from django.views.decorators.csrf import csrf_exempt
+# from rest_framework.views import APIView
+# from rest_framework.response import Response
+# from rest_framework_json_api.views import ModelViewSet
+import random 
+from rest_framework import generics
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from django.db.utils import IntegrityError
 
-# Create your views here.
+from .models import Restaurant
+from .serializers import RestaurantSerializer
+
+
+class GetAllRestaurantsView(APIView):
+
+    def get(self, request):
+        restaurants = Restaurant.objects.all()
+        serializer = RestaurantSerializer(restaurants, many=True)
+        return Response(serializer.data)
+    
+    def post(self, request):
+        restaurant_name = request.data.get('name')
+        try:
+            restaurant = Restaurant(name=restaurant_name)
+            restaurant.save()
+            return Response("Your restaurant was saved")
+        except IntegrityError:
+            return Response("You must provide a correct name for the restaurant", 400)
+
+
+class DeleteRestaurantView(APIView):
+
+    def delete(self, request, name):
+        try:
+            restaurant = Restaurant.objects.get(name=name)
+            restaurant.delete()
+            return Response("Your restaurant was deleted")
+        except Restaurant.DoesNotExist:
+            return Response("You can only delete an existing restaurant", 400)
+
+
+class GetRandomRestaurantView(APIView):
+
+    def get(self, request):
+        restaurants = Restaurant.objects.all()
+        restaurant = restaurants[random.randrange(len(restaurants))] 
```

Comment:
c'est du bonus mais je pense qu'il y'a des solutions pour récupérer un élem aléatoire via une query sql plutôt que de passer par python ce qui serait plus performant sur un large set de restos : https://stackoverflow.com/questions/962619/how-to-pull-a-random-record-using-djangos-orm